sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -1097,7 +1200,7 @@ class GpuRLikeMeta(
           GpuRLike(lhs, rhs, patternStr)
         }
         case StartsWith(s) => GpuStartsWith(lhs, GpuLiteral(s, StringType))
-        case Contains(s) => GpuContains(lhs, GpuLiteral(s, StringType))
+        case Contains(s) => GpuContains(lhs, GpuLiteral(UTF8String.fromString(s), StringType))
```

Comment:
A little confused, is it relevant to this PR on string contains ?

Reply:
Yes, this line fixed a test case failure: `test_rlike_multi_line`.
The framework will collect `GpuContains` but RLike generates a wrong `GpuContains`, the liternal should hold a UTF8String instead of Java String to keep consistent.

Reply:
You mean this is a Spark bug ? A UTF8String is expected but a String is given ?

Reply:
>You mean this is a Spark bug ?  

No, has nothing to do with Spark. 

> A UTF8String is expected but a String is given in Contains ?
Has nothing to do with Spark Contains.

This issue is only relevent with our GPU code.
By default, after `gpu overwrites`, GpuContains hold a literal which is a UTF8String.
But Rlike generate a GpuContains hold a literal which is a Java String. It's inconsistent behavior.

Reply:
It would be good to add a comment here. Since other expressions around (e.g. GpuStartsWith) have no such conversion.